marshal.go

```diff
@@ -144,3 +145,157 @@ func (i InstanceMetadata) MarshalXML(e *xml.Encoder, start xml.StartElement) err
 	// flush to ensure tokens are written
 	return e.Flush()
 }
+
+// startLocalName creates a start-tag of an XML element with the given local name and no namespace name.
+func startLocalName(local string) xml.StartElement {
+	return xml.StartElement{Name: xml.Name{Space: "", Local: local}}
+}
+
+type metadataMap map[string]string
+
+// MarshalXML is a custom XML marshaler for metadataMap, mapping each metadata name/value pair to a
+// correspondingly named XML element with the pair's value as character data content.
+func (m metadataMap) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+
+	for k, v := range m {
+		if err := e.EncodeElement(v, startLocalName(k)); err != nil {
+			return err
+		}
+	}
+
+	return e.EncodeToken(start.End())
+}
+
+// UnmarshalXML is a custom XML unmarshaler for metadataMap, mapping each XML element's name and
+// character data content to a corresponding metadata name/value pair.
+func (m metadataMap) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var v string
+	for {
+		t, err := d.Token()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if k, ok := t.(xml.StartElement); ok {
+			if err := d.DecodeElement(&v, &k); err != nil {
+				return err
+			}
+			m[k.Name.Local] = v
+		}
+	}
+	return nil
+}
+
+func metadataValue(i DataCenterInfo) interface{} {
+	if i.Name == Amazon {
+		return i.Metadata
+	}
+	return metadataMap(i.AlternateMetadata)
```

Comment:
Giving that we use either one or another and not both at a time, I'm happy with `alternate` then. Thanks for digging into that.